### ISSUE TITLE
Document safe passwords in manual/FAQ

### ISSUE THREAD

**Issue by tbm** [restic/restic#2238]
I wasn't sure how to pick a safe password, specifically how long it should be.

I [found this post on the forum](https://forum.restic.net/t/cryptographic-key-storage/1424/6) from @fd0 :

> Since we’re applying a key-derivation function (scrypt), a much shorter password (20-25 characters if you want to make sure) is plenty applying a key-derivation function (scrypt), a much shorter password (20-25 characters if you want to make sure) is plenty

Would it be possible to add this to the manual?  Probably the best would be to have a section on choosing password in the beginning.

It could also document some other questions that commonly pop up regarding password, e.g. you mention `--password-file <(gpg --decrypt restic-password.gpg)` in another forum post.

Furthermore, you [mention on the forum](https://forum.restic.net/t/what-if-keys-are-compromised/1474/2?u=fd0) that the data is encrypted with the "master keys":

> All data in the repo is encrypted (and signed) with the “master keys”. In order to decrypt the “master keys”, you need the content of a key file together with the correct password.

So what's the difference between the password and the master keys and where are the master keys stored?

In short: a section on password safety and management would be great.

Maybe the section could also mention why restic doesn't use GPG (I believe you explained that somewhere but I can't find the reference now).

Thank you

**Comment by fd0** [comment]
Great points, thanks for raising them!
